crates/ruff_server/src/edit/range.rs

```diff
@@ -0,0 +1,148 @@
+use super::{Document, PositionEncoding};
+use lsp_types as types;
+use ruff_source_file::OneIndexed;
+use ruff_source_file::{LineIndex, SourceLocation};
+use ruff_text_size::{TextRange, TextSize};
+
+/// Returns the [`TextRange`] for a LSP [`Range`] respecting the negotiated [`PositionEncoding`].
+pub(crate) fn text_range(
```

Comment:
Nit: We could introduce two new extension traits and expose this functionality as methods instead:

```rust
pub(crate) trait RangeExt {
	fn to_text_range(&self, text: &str, index: &LineIndex, encoding: PositionEncoding) -> TextRange{}
}

pub(crate) trait TextRangeExt {
	fn to_range(&self, text: &str, index; &LineIndex, encoding: PositionEncoding) -> types::Range {...}
}
```

It would slightly reduce the number of arguments that need to be passed. 

Another alternative is that we create our own `EditorRange` newtype wrapper around `types::Range` so that we can implement additional methods on it

```rust
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
pub(crate) struct EditorRange(types::Range);

impl EditorRange {
	pub(crate) fn from_text_range(range: TextRange, text: &str, index: &LineIndex, encoding: PositionEncoding) -> Self {
	...
  }

  pub(crate) fn to_text_range(text: &str, index: &LineIndex, encoding: PositionEncoding) -> TextRange {
		...
  }
}
```